### PR TITLE
Add support for RABBITMQ_CONF_ENV_FILE on Windows

### LINKED ISSUE #392: rabbitmq-env-conf.bat regression in 3.8.4
Reported here:

https://groups.google.com/d/topic/rabbitmq-users/VgdYACHQ9vM/discussion

### PR DESCRIPTION
**Pull request by lukebakken**
Fixes #392

### CODE REVIEW

**Comment by dumbbell** [comment]
On both Unix and Windows, `rabbitmq-env.conf`/`rabbitmq-env-conf.bat` is still loaded from the Bash/Batch scripts (from `rabbitmq-env{,.bat}` to be exact). This is still the case because some variables configure erl(1) flags which can't be configured at runtime.

I naively thought variables set in the Batch scripts would be inherited by the erl(1) child process, but perhaps this is not the case.

File: src/rabbit_env.erl [hunk 1]
```diff
@@ -1402,9 +1402,7 @@ load_conf_env_file(#{os_type := {unix, _},
                 true ->
                     case os:find_executable("sh") of
                         false -> Context1;
-                        Sh    -> do_load_conf_env_file(Context1,
-                                                       Sh,
-                                                       ConfEnvFile)
+                        Sh    -> do_load_conf_env_file(unix, Context1, Sh, ConfEnvFile)
```
Comment (dumbbell) [inline src/rabbit_env.erl:1405]: The OS type is available in `Context1`, you don't need to copy it to another argument.

File: src/rabbit_env.erl [hunk 2]
```diff
@@ -1490,18 +1494,54 @@ do_load_conf_env_file(Context, Sh, ConfEnvFile) ->
...
+           {"PLUGINS_DIR", get_default_plugins_path_from_env(Context)},
+           {"CONF_ENV_FILE_PHASE", "rabbtimq-prelaunch"}
+          ],
+
+    % rabbitmq/rabbitmq-common#392
+    % There is no way to double-quote the file path to deal with spaces because
+    % spawn_executable re-escapes them, resulting in the following:
+    % '\"C:\Users\lbakken\issues\misc space\rabbitmq-env-conf.bat\"' is not recognized as an internal or external command
+    % For now, assume users will not use paths with spaces.
+    Arg = io_lib:format("echo off && call ~ts && set", [ConfEnvFile]),
```
Comment (dumbbell) [inline src/rabbit_env.erl:1523]: Do you think we could use PowerShell instead, to solve escaping issues?

Comment (lukebakken) [reply src/rabbit_env.erl:1523]: I will try that. Great idea.

Comment (lukebakken) [reply src/rabbit_env.erl]: Actually I figured out the syntax to work with spaces - https://github.com/rabbitmq/rabbitmq-common/pull/393/commits/79b605ffbf2c80924f79b019fa194af6966b7364

File: src/rabbit_env.erl (hunk 1 above)
Comment (lukebakken) [reply src/rabbit_env.erl]: Done!

**Comment by dumbbell** [comment]
Thank you! I'm testing this this afternoon.

**Comment by dumbbell** [comment]
I'm trying on a Windows Server 2019 VM and `cmd.exe` is not on the path by default. I mean, if I type `cmd.exe`, it starts, but `C:\Windows\System32` is not in `%PATH%` and Erlang doesn't find it.

I will check a Windows Server 2012 VM to compare.

**Comment by dumbbell** [comment]
`C:\Windows\System32` is in `%PATH%` on Windows Server 2012 and Elrang finds `cmd.exe`.

**Comment by dumbbell** [comment]
The path to `cmd.exe` (by default) is available in the `%ComSpec%` environment variable. I will play with this a bit.

**Comment by lukebakken** [comment]
@dumbbell `os:find_executable` should be sufficient. Which Windows 2019 VM image are you using?

**Comment by dumbbell** [comment]
I'm using the official Windows Server 2019 image in Amazon EC2. I don't remember I tweaked the value of `%PATH%` except to add Erlang in it.

**Comment by lukebakken** [comment]
OK I will look for that for Vagrant

**Comment by lukebakken** [comment]
@dumbbell I started up [this box](https://app.vagrantup.com/StefanScherer/boxes/windows_2019) and `C:\Windows\System32` is in the `PATH`. It's safe to assume that `cmd.exe` will be in the path, and that `os:find_executable/1` can find it.

**Comment by lukebakken** [comment]
I built a Windows installer, and used the following `rabbitmq-env-conf.bat`:

```
set NODENAME=frazzle@%COMPUTERNAME%
set NODE_PORT=5680
```

Worked as expected!

**Pull request merged**